modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/introspection/enricher/ExtensionErrorsModelEnricher.java

```diff
@@ -0,0 +1,126 @@
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.extension.internal.introspection.enricher;
+
+import static java.lang.String.format;
+import static org.mule.runtime.api.i18n.I18nMessageFactory.createStaticMessage;
+import static org.mule.runtime.module.extension.internal.introspection.enricher.ExtensionErrors.CONNECTIVITY;
+import static org.mule.runtime.module.extension.internal.introspection.enricher.ExtensionErrors.RETRY_EXHAUSTED;
+import static org.mule.runtime.module.extension.internal.util.MuleExtensionUtils.getExtensionsErrorNamespace;
+
+import org.mule.runtime.api.exception.MuleRuntimeException;
+import org.mule.runtime.api.meta.model.ExtensionModel;
+import org.mule.runtime.api.meta.model.declaration.fluent.ExtensionDeclaration;
+import org.mule.runtime.api.meta.model.declaration.fluent.OperationDeclaration;
+import org.mule.runtime.api.meta.model.declaration.fluent.WithOperationsDeclaration;
+import org.mule.runtime.api.meta.model.operation.OperationModel;
+import org.mule.runtime.extension.api.annotation.error.ErrorTypeProvider;
+import org.mule.runtime.extension.api.annotation.error.ErrorTypes;
+import org.mule.runtime.extension.api.annotation.error.Throws;
+import org.mule.runtime.extension.api.declaration.DescribingContext;
+import org.mule.runtime.extension.api.declaration.spi.ModelEnricher;
+import org.mule.runtime.extension.api.error.ErrorTypeDefinition;
+import org.mule.runtime.extension.api.exception.IllegalModelDefinitionException;
+import org.mule.runtime.extension.api.model.property.ConnectivityModelProperty;
+import org.mule.runtime.module.extension.internal.introspection.describer.model.ExtensionElement;
+import org.mule.runtime.module.extension.internal.introspection.describer.model.MethodElement;
+import org.mule.runtime.module.extension.internal.introspection.describer.model.runtime.ExtensionTypeWrapper;
+import org.mule.runtime.module.extension.internal.introspection.describer.model.runtime.MethodWrapper;
+import org.mule.runtime.module.extension.internal.model.property.ImplementingMethodModelProperty;
+import org.mule.runtime.module.extension.internal.model.property.ImplementingTypeModelProperty;
+import org.mule.runtime.module.extension.internal.util.IdempotentDeclarationWalker;
+
+import java.util.Optional;
+import java.util.stream.Stream;
+
+/**
+ * {@link ModelEnricher} implementation which enriches the {@link ExtensionModel} and their {@link OperationModel}
+ * from the used {@link ErrorTypes} and {@link Throws} in an Annotation based extension.
+ *
+ * @since 4.0
+ */
+public class ExtensionErrorsModelEnricher implements ModelEnricher {
+
+  private ExtensionErrorModelFactory errorModelDescriber;
+
+  @Override
+  public void enrich(DescribingContext describingContext) {
```

Comment:
this logic belongs on the describer. The only thing that an enricher should do is adding the default error types (such as connectivity on connected operations). 

As this is today, there're pieces of this which apply to smart connectors but would never be because this code is annotation dependant